src/main/java/com/suse/saltstack/netapi/parser/JsonParser.java

```diff
@@ -84,8 +81,8 @@ public JsonParser(TypeToken<T> type) {
                 .registerTypeAdapter(LocalDateTime.class, new LocalDateTimeISOAdapter())
                 .registerTypeAdapter(ZonedDateTime.class, new ZonedDateTimeISOAdapter())
                 .registerTypeAdapter(StartTime.class, new StartTimeAdapter().nullSafe())
-                .registerTypeAdapter(Stats.class, new StatsDeserializer())
-                .registerTypeAdapter(Arguments.class, new ArgumentsDeserializer())
+                .registerTypeAdapter(Stats.class, new StatsAdapter().nullSafe())
```

Comment:
i don't think we need the `nullSafe` here. If something can be `null` is should generally marked with `Optional` then the `Optional` parser will take care of the null value.